app/controllers/api/dashboards_controller.rb

```diff
@@ -23,6 +23,15 @@ def index
       end
     end
 
+    if params.include?('sort') and (params['sort'].include?('user.role') or params['sort'].include?('user.name'))
```

Comment:
This implementation is not ideal. To fetch all the users from the User MS and updating all the dashboards with the current user info increases the request time in a way that is not negligible. It's also an unnecessary task.
Optimizing this process might require a bigger, more structural change, and its impacts should be evaluated.

Reply:
The alternative would be to have the user create/update process push it's data to this MS, which would either make those calls slower, or would have to be done in an async way, making error handling difficult and thus these results unreliable. This approach does increase the response time, but it's fairer in the sense that the user "paying" that price is the user using the expensive feature. It also results in a simpler, easier to debug flow, that doesn't depend on a complex solution to sync data on user create/update/delete.